agent/src/agentclient.rs

```diff
@@ -280,15 +284,17 @@ impl<T: APIServerClient> BrupopAgent<T> {
     async fn update_status_in_shadow(
         &self,
         bottlerocket_shadow: &BottlerocketShadow,
+        state: BottlerocketShadowState,
+        crash_count: u32,
+        state_transition_failure_timestamp: Option<DateTime<Utc>>,
```

Comment:
Something that might help with the proliferation of these arguments would be to encapsulate the error handling state in a struct, like `ShadowErrorInfo` or something like that. This could be something we implement the [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) trait on so that we don't need to use the constants `0` and `None` when constructing the Status.